### ISSUE TITLE
[expr.await] The lifetime of parameters and variables of the coroutine

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#4867]
[expr.await] p5
> Otherwise, control flow **returns** to the current coroutine caller or resumer ([dcl.fct.def.coroutine]) without exiting any scopes ([stmt.jump]).   

However, whether a variable with automatic storage duration is active or not is totally defined in [stmt.dcl] p3, which means "without exiting any scopes" seems to expound nothing. According to [basic.stc.auto] p1, the storage duration of the parameters and variables defined in a coroutine all have automatic storage duration, hence, their lifetimes are governed by [stmt.dcl] p3. Since what [expr.await] p5 says that 
> when a function returns, Q is after its body. 

the lifetime of the parameters of a coroutine has ended when the first suspension occurs (which **returns** to the caller or resume, that is, Q is after its body). However, it cannot interpret the lifetimes of these variables defined in the body, that is, a suspension may not end their lifetimes.   

If we admit the effect of "without exiting any scopes", which in turn, cannot interpret the lifetimes of the parameters of a coroutine, since the parameter scope does not exit out.  In conclusion, regardless of [stmt.dcl] p3 or "without exiting any scopes", neither of them can interpret both the lifetime of parameters of the coroutine and the variables in that coroutine.

**Comment by jensmaurer** [comment]
I'm hearing two concerns here:
- [stmt.dcl] p3 should tell the complete story, also covering destruction of coroutine parameters and non-destruction of local variables on suspension. Note that [stmt.dcl] p3 says "transfer of control within a function"; arguable suspension does not do that, but transfers control to outside the function.
- [expr.await] p5 "without exiting any scopes" does not relate to "active" from [stmt.dcl] p3, so we don't know what the former means.

**Comment by xmh0511** [comment]
> arguable suspension does not do that, but transfers control to outside the function.

Yes, I was talking about the transfer flow control that occurs after await-suspend evaluation.
